patchwork-fml/src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java

```diff
@@ -0,0 +1,375 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fml.common;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.StringJoiner;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.base.Preconditions;
+import cpw.mods.modlauncher.api.INameMappingService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.MarkerManager;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.launch.common.FabricLauncherBase;
+import net.fabricmc.mapping.tree.ClassDef;
+import net.fabricmc.mapping.tree.FieldDef;
+import net.fabricmc.mapping.tree.MethodDef;
+import net.fabricmc.mapping.tree.TinyTree;
+
+/**
+ * Some reflection helper code.
+ * This may not work properly in Java 9 with its new, more restrictive, reflection management.
+ * As such, if issues are encountered, please report them and we can see what we can do to expand
+ * the compatibility.
+ *
+ * <p>In other cases, accesssor mixins may be used.</p>
+ *
+ * <p>All field and method names should be passed in as intermediary names, and this will automatically resolve if Yarn mappings are detected.</p>
+ *
+ */
+public class ObfuscationReflectionHelper {
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final Marker REFLECTION = MarkerManager.getMarker("REFLECTION");
+	private static final TinyTree MAPPINGS = FabricLauncherBase.getLauncher().getMappingConfiguration().getMappings();
+	private static final String INTERMEDIARY = "intermediary";
+	private static final String NAMED = "named";
+	/**
```

Comment:
Blank lines before documentation comments
```suggestion

	/**
```